### PR TITLE
Copied over plugin

### PR DESCRIPTION
**Pull request by KenxinKun**
Copied original code into public repo

### CODE REVIEW

File: linting.mk [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# install a local golangci-lint if not found.
+$(GOLANGCI_TMP_BIN):
+	curl -OL https://github.com/golangci/golangci-lint/releases/download/v$(GOLANGCI_LINT_VERSION)/$(GOLANGCI_LINT_ARCHIVE)
+	mkdir -p $(GOLANGCI_DIR)/
+	tar -xf $(GOLANGCI_LINT_ARCHIVE) --strip-components=1 -C $(GOLANGCI_DIR)/
+	chmod +x $(GOLANGCI_TMP_BIN)
+	rm -f $(GOLANGCI_LINT_ARCHIVE)
+
+# Run golangci-lint inside the nexus image to simulate the CI and debug related errors.
+.PHONY: lint-ci
```
Comment (fabriziomoscon) [inline linting.mk]: remove this as it is 90poe specific with NEXUS

Comment (KenxinKun) [reply linting.mk]: removed the `lint-ci` section and related nexus variables

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+sonar.sources=.
```
Comment (fabriziomoscon) [inline sonar-project.properties]: as this is public, should sonar be used?
That would be only working if 90poe keep using it. Which will limit the open source repo longevity I think.

File: linting.mk [hunk 3]
```diff
@@ -0,0 +1,27 @@
+### linting options below
```
Comment (fabriziomoscon) [inline linting.mk:1]: Would it not be better to use bingo for this as it skips the need to have /tmp/90poe/ folder I guess?

File: README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# protoc-gen-gofullmethods
+
+protoc plugin for accessing the protoc-gen-go-grpc methods / URL paths
```
Comment (fabriziomoscon) [inline README.md]: I think that if we aim the upstream bug organisation to accept this as official plugin we should write a more detailed description of this plugin.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,31 @@
+include linting.mk
+
+export GO111MODULE=on
```
Comment (fabriziomoscon) [inline Makefile]: this has been default for a long time, perhaps we can omit it.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+.PHONY: ci
```
Comment (fabriziomoscon) [inline Makefile:3]: can we add a simple GHA workflow?

File: sonar-project.properties (hunk 2 above)
Comment (KenxinKun) [reply sonar-project.properties]: Indeed, removed sonar references

File: README.md (hunk 4 above)
Comment (KenxinKun) [reply README.md]: I've improved the description a bit and added a more practical example, that can be regenerated with the Makefile

File: CODEOWNERS [hunk 7]
```diff
@@ -1,2 +1 @@
 /CODEOWNERS @90poe/DevOps
```
Comment (fabriziomoscon) [inline CODEOWNERS]: also we don't need this file

File: Makefile (hunk 5 above)
Comment (KenxinKun) [reply Makefile]: removed!

File: CODEOWNERS (hunk 7 above)
Comment (KenxinKun) [reply CODEOWNERS]: removed

File: Makefile (hunk 6 above)
Comment (KenxinKun) [reply Makefile:3]: added very basic linting, testing and building

**Review by fabriziomoscon — APPROVED** [review]

**Review by sarunask — APPROVED** [review]

**Pull request merged**